opensearch-project/OpenSearch-Dashboards
# [OE] Adds dev doc script to precommit hook

### Description

Adds dev doc generation script to the precommit hook. It also validates that it is staged and fails if there is a diff and the diff isnt staged.

### Issues Resolved




## Screenshot



## Testing the changes



- Add a new Markdown file to the docs folder
- Try to commit that chnage without running `yarn docs:generateDevDocs`
- The commit should fail with the updated sidebar in the unstaged section
- Staging it should resolve the issue and allow you to commit the change

## Changelog

- chore: Adds a git pre commit hook to ensure that developer docs are always updated

### Check List
